### ISSUE TITLE
Really small docs issue in router section, no mention about leptos_router required features

### ISSUE THREAD

**Issue by itedya** [leptos-rs/leptos#2259]
**Describe the bug**
Documentation: `It’s easy to get started with the router.
First things first, make sure you’ve added the leptos_router package to your dependencies.`
That's absolutely true, but it would be nice for beginners to tell them about package features that they have to install.
Yes, I know - `console_error_panic_hook` describes everything in console dev tools tab, but it would be nice to tell the user beforehand.

**Leptos Dependencies**
None, it's documentation issue.

**To Reproduce**
1. Go to [Leptos docs - router section](https://book.leptos.dev/router/16_routes.html)

**Expected behavior**
I think it just should be written in docs.

**Screenshots**
With or without console_error_panic_hook - does not matter, everything compiles.
![obraz](https://github.com/leptos-rs/leptos/assets/69873350/706990f8-165f-40ed-9970-f0ed0730658e)

Error without console_error_panic_hook
![obraz](https://github.com/leptos-rs/leptos/assets/69873350/7cbf9813-a306-47de-9935-0360b14e29e8)

Error with console_error_panic_hook
![obraz](https://github.com/leptos-rs/leptos/assets/69873350/bd438c08-c6c6-4b86-a845-9149cbf3ad0b)

**Additional context**
None

**Comment by rakshith-ravi** [comment]
Fair point. I think it would make sense if we add a "debugging" section of sorts to the documentation where we inform people how to make panic hooks, logging, among other things, easier

**Comment by rakshith-ravi** [comment]
PR for this is welcome, btw!

CC @diversable

**Comment by gbj** [comment]
This is reasonable.

It is listed [in the actual docs for `leptos_router`](https://docs.rs/leptos_router/latest/leptos_router/#feature-flags), so I'm going to transfer this issue to the repo for the book.
